modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java

```diff
@@ -162,7 +162,7 @@ protected HttpRequest getHttpRequest(NHttpClientConnection conn, String path) {
         boolean forceContentLengthCopy = 
                 requestMsgCtx.isPropertyTrue(PassThroughConstants.COPY_CONTENT_LENGTH_FROM_INCOMING);
                             
-        if (forceContentLength) {
+        if (forceContentLength && needToProcessChunking) {
```

Comment:
Variable name 'needToProcessChunking' looks good. 
But still this gives this condition doesn't make sense to me. 
forceContentLength and needToProcessChunking should be true at the same time isn't it ?

Reply:
I think needToProcessChunking means need to process disable chunking.
In my opinion, the [processChunking function](https://github.com/wso2/wso2-synapse/blob/25bb3fc50575dec17d062906f45b843396be79b5/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java#L342) is a function that handles disable chunking.
If so, how about change the processChunking function name to processDisableChunking and replace needToProcessChunking with needToProcessDisableChunking?

Reply:
I think the variable name you have put is correct, however how we are using it inside getHttpRequest() method is not correct. 
Will you be able to explain more about the problem you are trying to solve. 
This area is very sensitive, so we need to check the every possible case.
Thanks.

Reply:
I think the actual meaning of the needToProcessChunking variable is not HTTP_1_0, it is not a GET request, nor a DELETE request with no payload. Therefore, needToProcessChunking means that the request has a Content-Length.
So, forceContentLength and needToProcessChunking can be true at the same time.